compliance/oscal-assessment-results.yaml

```diff
@@ -0,0 +1,1275 @@
+            - description: |
+                Result: satisfied
+              remarks: |
+                validate.msg: All pods have Istio sidecar proxy.
+                validate.exempt_namespaces_msg: Exempted Namespaces: istio-system, kube-system, uds-dev-stack, zarf
+          uuid: fc482a86-3801-474f-9c2d-38fe5eb0dd4f
+        - collected: 2024-06-30T22:27:27.821606552Z
+          description: |
+            [TEST]: fbe5855d-b4ea-4ff5-9f0d-5901d620577a - lula-validation-error
+          methods:
+            - TEST
+          relevant-evidence:
+            - description: |
+                Result: not-satisfied
+              remarks: |
+                Error getting Lula validation #fbe5855d-b4ea-4ff5-9f0d-5901d620577a: required domain is nil
+          uuid: 9a0794ac-4b32-4154-9694-974c4f26ddf9
+        - collected: 2024-06-30T22:27:27.827790559Z
+          description: |
+            [TEST]: ca49ac97-487a-446a-a0b7-92b20e2c83cb - enforce-mtls-strict
+          methods:
+            - TEST
+          relevant-evidence:
+            - description: |
+                Result: satisfied
+              remarks: |
+                validate.msg: All PeerAuthentications have mtls mode set to STRICT.
+          uuid: f1ec78f7-c262-4c92-bd4e-1ad2e8418d54
+        - collected: 2024-06-30T22:27:27.839654751Z
+          description: |
+            [TEST]: b0a8f21e-b12f-47ea-a967-2f4a3ec69e44 - gateway-configuration-check
+          methods:
+            - TEST
+          relevant-evidence:
+            - description: |
+                Result: not-satisfied
+              remarks: |
+                validate.msg: Expected Virtual Services not using expected Gateways
+                validate.msg_actual: Gateways configured: {"istio-admin-gateway/admin-gateway", "istio-passthrough-gateway/passthrough-gateway", "istio-tenant-gateway/tenant-gateway"}. VirtualServices using Gateways: {"istio-admin-gateway/admin-gateway": {"grafana-admin-grafana-80-grafana", "keycloak-admin-admin-access-with-optional-client-certificate", "neuvector-admin-neuvector-8443-neuvector-service-webui"}, "istio-passthrough-gateway/passthrough-gateway": set(), "istio-tenant-gateway/tenant-gateway": {"keycloak-tenant-public-auth-access-with-optional-client-certificate", "keycloak-tenant-remove-private-paths-from-public-gateway"}}
+                validate.msg_expected: Expected VirtualServices using Gateways: {"istio-admin-gateway/admin-gateway": {"grafana-admin-grafana-80-grafana", "keycloak-admin-admin-access-with-optional-client-certificate", "neuvector-admin-neuvector-8443-neuvector-service-webui"}, "istio-passthrough-gateway/passthrough-gateway": [], "istio-tenant-gateway/tenant-gateway": {"keycloak-tenant-emulate-gitlab-authorize-endpoint", "keycloak-tenant-emulate-gitlab-token-endpoint", "keycloak-tenant-emulate-gitlab-user-endpoint", "keycloak-tenant-public-auth-access-with-optional-client-certificate", "keycloak-tenant-remove-private-paths-from-public-gateway"}}
```

Comment:
Skimming this output and noticed this one - we recently removed the `keycloak-tenant-emulate-gitlab-*` virtualservices, I think those are the cause of the error here? Likely makes sense to update that list since we removed those entirely but also made me think of some other questions...
- Is that list hardcoded somewhere (just want to understand the effects of updating names/adding new virtualservices and where we may have to change also)?
- Should this validation allow for virtualservices not existing, as long as all the existent ones use the correct gateways?
- Maybe not today's problem but if someone deployed SWF/other tools on top of core would that fail these validations?

Reply:
This was a large discussion today actually - how flexible are the validations
- These are hardcoded - some rightfully so and others are still a WIP
- This identified example is one we are going to change now - it is TOO hardcoded

overall we need to get more feedback on what works and doesn't work. This example is detracting from greater compliance by being too specific. So we're updating some of our internal review guidelines (also WIP) to look for validations that cannot "scale" well as we put others apps on uds-core. 

We'll also get a fix in for the istio prometheus item you identified earlier. Thanks for helping us BEGIN to refine this process (and even how we review internally).

Reply:
Updated this control to capture higher-level intent of gateways and network flow separation. Easing the restrictions while still collecting evidence. This can be refined more in the future when we can put more guardrails in place. 

Also updated the prometheus validation and subsequently the threshold with 11 controls now satisfied.